### PR TITLE
rgw_sal_motr : [CORTX-30149] Fix for handling null version-id issues in multipart upload

### PR DESCRIPTION
**Pull request by priyanka25081999**
Problem Statement:
Multipart-upload is not working as expected for null version-id scenarios.

Solution:
To handle the null version-id cases in multipart upload API, created null reference key(obj/null) at the time of object PUT(same way as we used in case of normal obj)

Signed-off-by: Priyanka Salunke [priyanka.s.salunke@seagate.com]

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by cdeshmukh** [comment]
>Solution:
To handle null version-id cases, added the code in MotrMultipartUpload::complete() function.
Also, re-use the functions to create an extra null entry (obj/null) for an object from PR https://github.com/Seagate/cortx-rgw/pull/200
solution should just tell in plain english about the fix, without giving actual code details like function names, plz update

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2730,6 +2730,13 @@ int MotrObject::update_null_reference(const DoutPrefixProvider *dpp, rgw_bucket_
   rgw_bucket_dir_entry current_null_key_ref;
   current_null_key_ref.key.name = this->get_name();
   current_null_key_ref.key.instance = this->get_instance();
+
+  // Set the key and instance for multipart object from ent structure
+  if (ent.meta.category == RGWObjCategory::MultiMeta)
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:2735]: well why not putting lines 2371-2732 & 2737-2738 in if-else condition

Comment (priyanka25081999) [reply src/rgw/rgw_sal_motr.cc:2733]: sure, done.

**Review by shriya-deshmukh — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]
